### PR TITLE
Ensure that absent bone names work

### PR DESCRIPTION
**Pull request by appgurueu**
Achieved by refactoring the code to use `std::optional<std::string>` instead of `core::stringc`.

[Soft prerequisite](https://github.com/JosiahWI/irrlicht/pull/14#discussion_r1470421154) for gltf support: gltf does not require bones to be named. We want to support models with "unnamed" bones.

Corresponding Minetest changes: https://github.com/minetest/minetest/pull/14330

### How to test

1. Read carefully (the changes should be pretty straightforwardly fine "per construction")
2. Regression tests: Do .b3d models like Sam still work? .x models (like our "cool guy" in the demo)? Attachments to bones?

![cool guy :\] haha](https://github.com/minetest/irrlicht/assets/34514239/3049e2a6-3f61-49a3-9db4-201e27777d79)

### CODE REVIEW

**Comment by sfan5** [comment]
Wondering why you didn't just make it a `std::string`?

**Comment by appgurueu** [comment]
> Wondering why you didn't just make it a `std::string`? or is `""` somehow reserved?

Two reasons:

1. I'm not sure whether the empty string would be an admissible bone / scene node name. I don't see a definitive reason why it shouldn't work, at least. If it is, I don't want to mess with that.
2. `std::optional` forces me to handle the "absence" case everywhere. It makes it explicit when a name is absent, forcing me to handle that correctly, whereas "empty name means absence" is just a convention that's easy to (accidentally) violate.

File: include/ISceneNode.h [hunk 1]
```diff
@@ -117,23 +118,14 @@ namespace scene
...
-		virtual void setName(const c8* name)
-		{
-			Name = name;
-		}
-
-
 		//! Sets the name of the node.
 		/** \param name New name of the scene node. */
-		virtual void setName(const core::stringc& name)
+		virtual void setName(const std::optional<std::string> &name)
```
[the remark this answers is not in the document]
Comment (appgurueu) [reply include/ISceneNode.h:128]: I think that'd just be moving part of the `optional` interface to the containing class.

[the remark this answers is not in the document]
Comment (sfan5) [reply include/ISceneNode.h:128]: indeed. looks like you chose not to change it, that's ok.

**Pull request merged**